### ISSUE TITLE
Compiling with Cygwin

### ISSUE THREAD

**Issue by marnunez** [wnd-charm/wnd-charm#42]
Hi! I successfully compiled both wnd-charm 1.52.775 and the current master branch under Cygwin 1.7.33 x64 on a Windows 7 x64 machine.
Both `libfftw3-dev`, `libtiff-dev` are installed and current, 

However, I'm getting these errors:
```
Processing directory '/cygdrive/c/asdfasdf/asdfasdf/gfgfgd/asasd/qweqwe/asdfAFR'
assertion "algorithm != NULL && "algorithm not found when calling > > > > >
FeatureNames::getGroupByName"" failed: file "FeatureNames.cpp", line 330, function: static const
FeatureGroup* FeatureNames::getGroupByName(const string&)
Aborted (`core' generado)
```
I've also tried the Windows binaries + libraries supplied in http://ome.grc.nia.nih.gov/wnd-charm/, but seem really old and get stuck at "start processing image..."

Any hints?

Thanks in advance

**Comment by igg** [comment]
You could try setting verbosity up to 9 (-s9 flag) to get a better idea where the fault is.  It is almost certainly due to a failure in loading/registering the feature or transform algorithms "pre-main".  Unfortunately, we have no Windows systems here to even try to reproduce this.  It works on two different OSes (Linux & OS X) and two different compiler systems (gcc, clang), so cygwin is definitely the odd man out here.  It would be nice to support cygwin and Windows of course, but we really don't have the resources to support everything, and Windows is a lower priority as a compute & development platform.  Its probably not helpful, but why not run Windows as a guest OS in VirtualBox on Linux?  That way you have the best of both worlds.
-I

**Comment by marnunez** [comment]
Thanks for your reply!

Sadly, the option of using a Linux machine is not available for me, institutional policies (don't even get me started!), the best I can do is the opposite: a Linux VM running on VirtualBox for Windows. I do have that, but since all my scripts are already running in Windows, having a VM just for wnd-charm didn't seem the most elegant solution. That's what got me to try and compile the binaries for either:

- Native Windows using mingwin
- Windows + Cygwin using cygwin

That way, I could use the binaries with the scripts I already have. Not to be stubborn, but would you happen to know in which conditions were those old binaries compiled? Any special flags?

**Comment by igg** [comment]
I fully understand - I have plenty of institutional policies to deal with myself!
The old binaries were compiled using Microsoft's development tools, but I don't have the details - it was work done many years ago by Lior Shamir who is now at Lawrence Tech.  The software has changed quite radically since those days, so whatever was done then will not be applicable to the code as it is today. Using MS's standard tools is certainly worth a try though.

I'm assuming you've done small-scale trials and have a good idea of what this will entail when you scale it up for production.  Its just that when we use this for real work in the lab (which we do constantly), its never with a desktop OS. I'm worried for example, if the file system being used is sufficiently compliant with POSIX-style file locking to enable running multiple wndchrm instances in parallel.  I'm just wondering if this pre-main initialization problem (if that's what it is) could potentially be only the tip of an iceberg.

Back to the issue at hand, did you get any output with more verbosity using the -s flag?  If its a pre-main initialization problem, it will fail very early, and this can't be debugged with command-line prompts (since we're looking at stuff that happens before main executes).  My own fork (https://github.com/igg/wnd-charm) will at least let you see what gets initialized when you set the WNDCHRM_VERBOSITY environmental variable.
-I
